File: C++/프로그래머스/0/120846. 합성수 찾기/합성수 찾기.cpp
#include <string>
#include <iostream>

using namespace std;

bool is_prime(int n) {
    int num = n;
    
    for (int i = 2; i < n; i++) {
        while (num % i == 0) num /= i;
    }
    
    return num > 1;
}

int solution(int n) {
    int answer = 0;
    
    for (int i = 4; i <= n; i++) {
        if(!is_prime(i)) {
            answer++;
        }
    }
    
    return answer;
}

File: Go/프로그래머스/0/120846. 합성수 찾기/합성수 찾기.go
func is_prime(n int) bool {
    num := n
    for i := 2; i < n; i++ {
        for num % i == 0 {
            num /= i
        }
    }
    return num > 1
}

func solution(n int) int {
    count := 0
    
    for i := 2; i <= n; i++ {
        if !is_prime(i) {
            count++
        }
    }
    
    return count
}